include/sdf/system_util.hh

```diff
@@ -26,7 +26,7 @@
  */
 
 #if defined _WIN32 || defined __CYGWIN__
-  #ifdef BUILDING_DLL
+  #ifdef BUILDING_SDFORMAT
```

Comment:
The else clause will be chosen when building a static `sdformat` library.
This will cause warnings/errors to be emitted, because `__declspec(dllimport)` is used, while no dll is actually used/built.
The logic should be something like the following:
```c++
#if defined(_WIN32)
#ifdef BUILDING_SDFORMAT
#define API __declspec(dllexport)
#elif !defined(SDFORMAT_STATIC)
#define API __declspec(dllimport)
#else
#define API
#endif
#else
// non-windows stuff here
#endif
```

Reply:
@madebr, I fixed the macro conditionals. I'm not sure what you mean by 
```cmake
if(NOT BUILD_SHARED_LIBS)
target_compile_definitions(${_name} PUBLIC BUILDING_SDFORMAT)
endif()
```

Reply:
You mean
```cmake
if(NOT BUILD_SHARED_LIBS)
target_compile_definitions(${_name} PUBLIC SDFORMAT_STATIC)
endif()
```

Reply:
Yes, the static definition is missing.
In the current pr state:
```cmake
if(NOT BUILD_SHARED_LIBS)
    target_compile_definitions(${_name} PUBLIC SDFORMAT_STATIC_DEFINE)
endif()
```

Reply:
Done